apps/b2b/components/Auth/RegistrationForm.tsx

```diff
@@ -0,0 +1,124 @@
+import { gql, useMutation } from '@apollo/client'
+import { yupResolver } from '@hookform/resolvers/yup'
+import ErrorMessage from 'components/ErrorMessage'
+import Input from 'components/Input'
+import Link from 'next/link'
+import { useForm } from 'react-hook-form'
+import { Button, Typography } from 'ui'
+import * as yup from 'yup'
+
+const FormSchema = yup.object({
+  name: yup.string().required().label('Name'),
+  email: yup.string().email().required().label('Email'),
+  password: yup.string().required().label('Password'),
+  confirm_password: yup
+    .string()
+    .required()
+    .oneOf([yup.ref('password'), null], "Password doesn't match")
+    .label('Confirm Password'),
+})
+
+type FormInputs = yup.InferType<typeof FormSchema>
+
+const RegistrationForm = () => {
+  const [registerUser, { loading }] = useMutation(REGISTER_USER)
+  const {
+    register,
+    handleSubmit,
+    formState: { errors, isSubmitting, isSubmitted },
+  } = useForm<FormInputs>({
+    resolver: yupResolver(FormSchema),
+  })
+
+  const onSubmit = handleSubmit((data: FormInputs) => {
+    const { name, email, password } = data
+
+    registerUser({
+      variables: {
+        objects: [
+          {
+            name,
+            email,
+            user_credential: { data: { identifier: email, password } },
+          },
+        ],
+      },
+    })
+  })
+
+  if (!!errors && isSubmitted) {
```

Comment:
no need to do double negation as you are already using an AND operator so the expression result will always be boolean.  

`if (errors && isSubmitted)`